bootstrapper/internal/kubernetes/k8sapi/kubectl/kubectl.go

```diff
@@ -11,110 +11,96 @@ import (
 	"errors"
 	"fmt"
 
-	"github.com/edgelesssys/constellation/v2/internal/kubernetes"
 	corev1 "k8s.io/api/core/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsclientv1 "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/retry"
 )
 
 // ErrKubeconfigNotSet is the error value returned by Kubectl.Apply when SetKubeconfig was not called first.
 var ErrKubeconfigNotSet = errors.New("kubeconfig not set")
 
-// Client wraps marshable k8s resources into resource.Info fields and applies them in a cluster.
-type Client interface {
-	// ApplyOneObject applies a k8s resource similar to kubectl apply.
-	ApplyOneObject(info *resource.Info, forceConflicts bool) error
-	// GetObjects converts resources into prepared info fields for use in ApplyOneObject.
-	GetObjects(resources kubernetes.Marshaler) ([]*resource.Info, error)
-	CreateConfigMap(ctx context.Context, configMap corev1.ConfigMap) error
-	AddTolerationsToDeployment(ctx context.Context, tolerations []corev1.Toleration, name string, namespace string) error
-	AddNodeSelectorsToDeployment(ctx context.Context, selectors map[string]string, name string, namespace string) error
-	// WaitForCRD waits for the given CRD to be established.
-	WaitForCRD(ctx context.Context, crd string) error
-	ListAllNamespaces(ctx context.Context) (*corev1.NamespaceList, error)
-}
-
-// clientGenerator can generate new clients from a kubeconfig.
-type clientGenerator interface {
-	NewClient(kubeconfig []byte) (Client, error)
-}
-
-// Kubectl implements kubernetes.Apply interface and acts like the Kubernetes "kubectl" tool.
+// Kubectl implements acts like the Kubernetes "kubectl" tool.
 type Kubectl struct {
-	clientGenerator
-	kubeconfig []byte
+	kubernetes.Interface
+	apiextensionClient apiextensionsclientv1.ApiextensionsV1Interface
+	builder            *resource.Builder
 }
 
-// New creates a new kubectl using the real clientGenerator.
+// New returns an empty Kubectl client. Need to call Initialize before usable.
 func New() *Kubectl {
-	return &Kubectl{
-		clientGenerator: &generator{},
-	}
+	return &Kubectl{}
 }
 
-// Apply will apply the given resources using server-side-apply.
-func (k *Kubectl) Apply(resources kubernetes.Marshaler, forceConflicts bool) error {
-	if k.kubeconfig == nil {
-		return ErrKubeconfigNotSet
+// Initialize sets sets all required fields so the Kubectl client can be used.
+func (k *Kubectl) Initialize(kubeconfig []byte) error {
```

Comment:
I tried to do that, but we need dependency injection for the Kubewrapper tests. If we were to initialize the kubectl object [here](https://github.com/edgelesssys/constellation/blob/main/bootstrapper/internal/kubernetes/kubernetes.go#L155), the tests would expect a file on the filesystem. This seemed like an antipattern to me. I nonetheless feel like it would improve the overall code quality.

Reply:
Maybe replacing the `client` variable with a `newClient` function might do the trick.
Could also be somewhat unwieldy.
Maybe keeping it as is works, while keeping in mind that we want to refactor the entire package in the future, especially since we moved a large part of it to helm